drs_cli/client.py

```diff
@@ -193,3 +213,15 @@ def _get_headers(self):
         }
         if self.token:
             self.headers['Authorization'] = 'Bearer ' + self.token
+
+
+def _check_drs_regex(uri) -> bool:
```

Comment:
I would probably merge the two methods to check and parse into a single method that parses and returns host and object if it can and raises an error otherwise. Then in the main code call the function and assign to a tuple as you do for the `_get_host_and_id()` function, but wrap it in a `try`/`except` block

Reply:
Should I leave this as it was before (two separate methods), coz I have added regex for HTTP, HTTPS URIs, and also regex for DRS IDs? It improves the readability of the code.

Reply:
And the `if` conditions are easier to write when the methods are separate. 😅

Reply:
I implemented this now such that checking and parsing is done in one go, though separately for extracting the host and the object ID.